engine/src/main/java/org/terasology/rendering/dag/FBOStateChange.java

```diff
@@ -15,19 +15,18 @@
  */
 package org.terasology.rendering.dag;
 
-import java.util.HashMap;
-import static org.lwjgl.opengl.GL11.GL_FILL;
-
 /**
  * TODO: Add javadocs
  */
-class DefaultStateChangeStorage extends HashMap<Class<? extends StateChange>, StateChange> {
+public class FBOStateChange extends AbstractStateChange<Integer> {
+    private static FBOStateChange defaultInstance = new FBOStateChange(0);
 
-    DefaultStateChangeStorage() {
-        // TODO: add more default values of StateChange's here
-        // TODO: be sure every custom state change has a default value here
+    FBOStateChange(Integer value) {
+        super(value);
+    }
```

Comment:
Actually, in line with what I wrote below regarding readability and the SetWireframe state change, BindFBO could simply take the fbo name (a string) in input and convert it to an opengl id only WHEN and IF it deals with the task creation/update.

On the broader picture, this seems to hint that rather having a map StateChange.class -> RenderTaks.class held in the RenderTaskListGenerator, each StateChange implementation could be made smarter and be the one with a 

`StateChange.generateTask()`

method, invoked by the task list generator if appropriate for that state change. This would also give the chance to the state change to wire itself to the task.